core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -386,4 +393,20 @@ public Configuration get() {
       return conf;
     }
   }
+
+  private static class EmptySerializedConfiguration implements Serializable {
+
+    private transient volatile Configuration conf = null;
+
+    public Configuration get() {
+      if (conf == null) {
+        synchronized (this) {
```

Comment:
Is this a valid situation? We might want to throw an exception here, since this will be an invalid configuration object in this case.

Reply:
No, that's a good point, it should throw an exception. This `get()` method was mistakenly accessed during split generation part, where the manifest files are being read even though the config has already been removed earlier by calling the `SerializableTable.copyOf(table)` method.